### ISSUE TITLE
Timezone support

### ISSUE THREAD

**Issue by deadcoder0904** [h-dong/Serina#59]
See slide no 25 & 26 -> https://growth.design/case-studies/superhuman-user-onboarding/

![timezone](https://user-images.githubusercontent.com/16436270/129709372-8ac1f7a6-6069-4c1d-b4e9-61dd5823b1ea.png)

I think the Superhuman event parser is the best thing out there so a copy of that would be awesome.

Are you still planning to complete the milestones or working on other projects?

I think this is the best NLP event parser library out there in JavaScript other than [Sherlock](https://github.com/neilgupta/Sherlock) :)

**Comment by h-dong** [comment]
Hi @deadcoder0904, thanks for leaving this issue. Sorry for the 1 year late reply, I must have missed the notification 😅 To speak the truth, I didn't think anyone else is using this.

> I think the Superhuman event parser is the best thing out there so a copy of that would be awesome.

I didn't think about timezones when I started this project, but now I think about it this is a great suggestion. I'm going to modify the milestone list later as I have some ideas about this project, and I'll make sure to add this in. Thanks!

> Are you still planning to complete the milestones or working on other projects?

I'm not actively working on this as you can probably tell, but I would like to come back to this at some point. I have another personal project dependent on this, but it is hard to say when that'll be. Knowing there are people out there using this library other than me does give me some motivation to come back to this sooner!

----

In the slim chance you see this, are you still using this library? Would love some more feedback 😄

**Comment by deadcoder0904** [comment]
@h-dong i am not using this either lol. i abandoned that project but will come back to it & will have multiple saas in future that need a scheduler so will use this library or something else.

sherlock.js is another alternative that is decent.

**Comment by h-dong** [comment]
Haha no worries, I will see if I can find some time and work on this at some point. Need to think about the time zone and how to best fit it in this library. Also if you have any other feedback feel free to share them 🙂

**Comment by h-dong** [comment]
Hi @deadcoder0904, not sure if this is something you still need or interested in. Either way, I've been doing some thinking about this and it is a bit more complex than I thought:

Date time works either in UTC or local timezone, creating a date object from scratch based on a given timezone isn't actually supported natively. I've removed Luxon as part of v2 refactor, so that made it a lot more complicated to work with timezones. The best way if not only way to create a new date is to create a date in UTC and then apply timezone offset. This not only need a list of all the timezone and their +/- hours relative to UTC, but with light savings this will likely to change throughout the year for some regions. Long story short, it gets pretty nasty.

In the superhuman interface they are only dealing with time related input, which is not the case with Serina. This library focus on finding the time date related keywords and then parse them. So adding in timezone logic isn't as simple as adding a new filter, but more like adding logic on all filters. So that adds complexity too.

Bottom line is I'm still thinking about this, struggling to find a neat way to integrate timezone into Serina at the moment. This means it might not make its way to Serina v2. Any suggestions are welcome!

**Comment by h-dong** [comment]
I think might wait for Temporal to be added, it would make the timezone feature a lot simpler https://tc39.es/proposal-temporal/docs/.
